tests/lib/rules/no-get-with-default.js

```diff
@@ -0,0 +1,38 @@
+const rule = require('../../../lib/rules/no-get-with-default');
+const RuleTester = require('eslint').RuleTester;
+
+const ruleTester = new RuleTester({
+  parserOptions: {
+    ecmaVersion: 2015,
+    sourceType: 'module',
+  },
+});
+
+ruleTester.run('no-get-with-default', rule, {
+  valid: [
+    "const test = this.get('key') || [];",
+    "const test = get(this, 'target') || [];",
+    "testClass.getWithDefault('key', [])",
+    "getWithDefault.testMethod(testClass, 'key', [])",
+  ],
+  invalid: [
+    {
+      code: "const test = this.getWithDefault('key', []);",
+      errors: [
+        {
+          message: "Use `this.key` instead of `this.getWithDefault('key', ...)`",
```

Comment:
Wouldn't importing the error message function from the rule file not be testing anything? I thought about doing at first but didn't because of that reason. ie. that would be saying
`getErrorMessage('a', true) === getErrorMessage('a', true)`

Reply:
Another benefit is that this error message is actually readable, which IMO is more important than being DRY in this use case.

But I'm open =)

Reply:
That's fair, then it's fine to leave as is this in your case. For static string error messages, I typically prefer to use a variable instead of duplicating the string for dozens of test cases, so that error message tweaks only need to be made in one place.